### DISCUSSION TITLE
CORE-ESP32 template + WS2812B

### DISCUSSION THREAD

**Discussion by adr-bed** [arendst/Tasmota#17520] [Q&A] [answered]
Hello
I have configured a [CORE-ESP32 board](https://templates.blakadder.com/luatos_CORE-ESP32.html) using this template:

`{"NAME":"Luatos CORE-ESP32","GPIO":[1,1,1,1,1,1,1,1,1,1,1,0,544,416,0,0,0,0,1,1,1,1],"FLAG":0,"BASE":1}`

![CORE-ESP32 Board - Configure Template](https://user-images.githubusercontent.com/91057963/209681399-4ceacbcd-5016-4ac9-8cab-cc757eade17c.png)

This give me a nice control over the second onboard LED using PWM (dimmer + toggle button):

![CORE-ESP32 Board - Main Menu 0](https://user-images.githubusercontent.com/91057963/209681755-7b568cdc-ae15-4fdc-b921-2979e4c80560.png)

The problem is when I configure additional light (WS2812B strip) using GPIO7 as WS2812, the main menu page shows only controls for this light (3 channels + toggle button), but no longer for onboard LED:

![CORE-ESP32 Board - Main Menu](https://user-images.githubusercontent.com/91057963/209681346-b729b9c8-36ae-4142-9d7a-4aefe9684418.png)

Could someone please explain me what I'm missing?

**Comment by sfromis** [comment, marked as answer]
The lights feature of Tasmota only supports *either* PWM *or* neopixels, not both at the same time. With ESP32, you still have the option of using the Berry language to control neopixels and/or PWM.

By configuring neopixels as WS2812 channel 2 and up (to dissociate from being seen as a light), you can control the [neopixels using Berry code](https://tasmota.github.io/docs/Berry_Addressable-LED/).

Likewise you can config PWM on channels 6 and up (instead of using 1-5 which are for lights), and then control [PWM using Berry code](https://tasmota.github.io/docs/Berry/#gpio-module).

In both cases, you can also use the features of Berry to do stuff like creating commands and/or web UI controls, but things take time.....

**Comment by adr-bed** [comment]
All clear, thank you!
Even if the PWM control is now missing on the main menu, is it safe to leave the template like this?

**Reply by sfromis** [reply]
I'd be surprised if not, but I'd prefer to only have useful stuff configured. Reduces confusion risks later.....
